homeassistant/components/tuneblade_remote/media_player.py

```diff
@@ -0,0 +1,161 @@
+"""TuneBlade Remote media player entity."""
+
+import asyncio
+import logging
+from typing import Any
+
+from homeassistant.components.media_player import (
+    MediaPlayerEntity,
+    MediaPlayerEntityFeature,
+    MediaPlayerState,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import DOMAIN, MASTER_ID
+from .coordinator import TuneBladeDataUpdateCoordinator
+from .entity import TuneBladeEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up TuneBlade Remote from a config entry."""
+    coordinator: TuneBladeDataUpdateCoordinator = config_entry.runtime_data
+
+    runtime_data = hass.data.setdefault(DOMAIN, {}).setdefault(
+        config_entry.entry_id, {}
+    )
+    added_ids = runtime_data.setdefault("added_ids", set())
+
+    entities_master = []
+    entities_others = []
+
+    for device_id, device_data in coordinator.data.items():
+        if device_id == MASTER_ID:
+            entities_master.append(
+                TuneBladeMediaPlayer(coordinator, device_id, device_data)
+            )
+            added_ids.add(device_id)
+
+    for device_id, device_data in coordinator.data.items():
+        if device_id != MASTER_ID:
+            entities_others.append(
+                TuneBladeMediaPlayer(coordinator, device_id, device_data)
+            )
+            added_ids.add(device_id)
+
+    if entities_master:
+        async_add_entities(entities_master, True)
+
+    if entities_others:
+        async_add_entities(entities_others, True)
+
+
+class TuneBladeMediaPlayer(TuneBladeEntity, MediaPlayerEntity):
+    """Media player for TuneBlade devices."""
+
+    def __init__(
+        self,
+        coordinator: TuneBladeDataUpdateCoordinator,
+        device_id: str,
+        device_data: dict[str, Any],
+    ) -> None:
+        """Initialize the media player entity."""
+        super().__init__(
+            coordinator,
+            coordinator.config_entry,
+            device_id,
+            device_data.get("name"),
+        )
+        self.device_id = device_id
+        self._attr_name = device_data["name"]
```

Comment:
all new integrations should use `_attr_has_entity_name = True`, so make sure you set the device name to the home assistant device name, and set `_attr_name = None` to make HA take the device name as entity name